pygeo/DVGeometry.py

```diff
@@ -416,16 +438,51 @@ def addRefAxis(self, name, curve=None, xFraction=None, volumes=None,
             # Loop through sections and compute node location
             place = 0
             for j, vol in enumerate(volOrd):
+                # sectionArr: indices of FFD points grouped by section
                 sectionArr = numpy.rollaxis(lIndex[vol], alignIndex, 0)
                 skip = 0
                 if j > 0:
                     skip = 1
                 for i in range(nSections[j]):
-                    LE = numpy.min(self.FFD.coef[sectionArr[i+skip,:,:],0])
-                    TE = numpy.max(self.FFD.coef[sectionArr[i+skip,:,:],0])
-                    refaxisNodes[place+i,0] = xFraction*(TE - LE) + LE
-                    refaxisNodes[place+i,1] = numpy.mean(self.FFD.coef[sectionArr[i+skip,:,:],1])
-                    refaxisNodes[place+i,2] = numpy.mean(self.FFD.coef[sectionArr[i+skip,:,:],2])
+                    # getting all the section control points coordinates
+                    pts_tens = self.FFD.coef[sectionArr[i + skip, :, :], :]  # shape=(A,B,3)
+                    # reshaping into vector to allow rotation (if needed)
+                    pts_vec = numpy.copy(pts_tens.reshape(numpy.shape(pts_tens)[0] * numpy.shape(pts_tens)[1], 3))  # shape=(A*B,3)
+
+                    if rotType == 0 and rot0ang:
+                        # rotating the FFD to be aligned with main axes
+                        for ct_ in range(numpy.shape(pts_vec)[0]):
+                            # here we loop over the pts_vec, rotate them and insert them inplace in pts_vec again
+                            p_ = numpy.copy(pts_vec[ct_ , :])
```

Comment:
why is the variable p_ defined? Why do we need to use numpy.copy here?

Reply:
Here I am trusting Mads/Sandy/Nick(?) fix... I know python can be tricky because of the "pointer/variable" approach, so by default I am fine with staying on the safe side and adding an extra `numpy.copy`. Anyway, I will try and see if removing this has any effect!